core/src/test/java/org/infinispan/configuration/ConfigurationUnitTest.java

```diff
@@ -52,6 +54,20 @@ public void testBuild() {
    }
 
    @Test
+   public void testDocumentationConfigs() throws IOException {
+      //tag::basic-example[]
+      EmbeddedCacheManager manager = new DefaultCacheManager("configs/my-custom-config.xml");
+      Cache defaultCache = manager.getCache();
+      //end::basic-example[]
+      //tag::named-cache-retrieval[]
+      Cache someNamedCache = manager.getCache("local");
+      //end::named-cache-retrieval[]
+
+      EmbeddedCacheManager manager2 = new DefaultCacheManager("configs/my-custom-config2.xml");
+      EmbeddedCacheManager manager3 = new DefaultCacheManager("configs/my-custom-config3.xml");
```

Comment:
Seems to me you might be leaking cache managers here? They are not being shutdown...

I know is for documentation, but in the past I've had issues with creating cache managers without using the `TestCacheManagerFactory` methods. These issues were the result of the cache managers not being isolated in our concurrent test suite. 

Not sure if you'd have problems too, but I'd favour such examples to be somewhere else where maybe tests that verify code examples can run sequentially, where you don't run such risks.